esbulk.go

```diff
@@ -0,0 +1,200 @@
+package esbulk
+
+import (
+	"bufio"
+	"compress/gzip"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+)
+
+// Options represents bulk indexing options.
+type Options struct {
+	Servers     []string
+	Index       string
+	Purge       bool
+	Mapping     string
+	DocType     string
+	NumWorkers  int
+	ZeroReplica bool
+	GZipped     bool
+	BatchSize   int
+	Verbose     bool
+	IDField     string
+	Scheme      string // http or https; deprecated, use: Servers.
+	Username    string
+	Password    string
+	MaxRetries  int
+}
+
+// ProcessLDJFile reads input file and creates an index given options using
+// multiple workers
+func ProcessLDJFile(r io.Reader, options Options) (int, error) {
+	count := 0
+
+	if options.Index == "" {
+		return count, errors.New("index name required")
+	}
+
+	if options.Verbose {
+		log.Println(options)
+	}
+
+	if options.Purge {
+		if err := DeleteIndex(options); err != nil {
+			return count, err
+		}
+		time.Sleep(5 * time.Second)
```

Comment:
Maybe add a comment why the `time.Sleep` is needed?

Reply:
This is something I just moved and there is not much description from [author](https://github.com/miku/esbulk/commit/1f4bfcdeb6dc496390e9bda3174cd5312d591dbd). I would personally either change that to solid solution (for example wait and check index until it is really deleted with some timeout of leave it like that). WDYT?

Reply:
Ok I rewrite this fragile sleep into using logic for checking if index exists and do a few retries.